### PR TITLE
Revert "Remove init of pylogger.rs from python code"

### PR DESCRIPTION
**Pull request by peterschwarz**
This reverts commit ba9517ec0937e0674ee561fdd6ee79d31d004f8c.

This restores proper logging in the rust code.  With the removal of this
init of pylogger, log messages from the lower layer (i.e. libsawtooth)
are not included in the logs.  This makes debugging difficult.

Signed-off-by: Peter Schwarz <pschwarz@bitwise.io>

### CODE REVIEW

**Comment by peterschwarz** [comment]
Added a commit to fix rust lint

**Comment by peterschwarz** [comment]
Added a commit to fix the doc issues

File: validator/src/pylogger.rs [hunk 1]
```diff
@@ -42,12 +42,12 @@ pub fn set_up_logger(verbosity: u64, py: Python) {
...
 #[no_mangle]
 #[allow(unused)]
-#[allow(private_no_mangle_fns)]
 pub extern "C" fn pylogger_init(verbosity: usize) {
     let gil = Python::acquire_gil();
     let py = gil.python();
-    PyLogger::init(determine_log_level(verbosity as u64), py)
-        .expect("Failed to initialize PyLogger");
+    if let Err(_) = PyLogger::init(determine_log_level(verbosity as u64), py) {
+        warn!("Attempted to initialize logger twice; ignoring");
```
Comment (vaporos) [inline validator/src/pylogger.rs]: Is that the only way PyLogger::init() can fail? If not, should the error text be logged?

File: validator/src/pylogger.rs [hunk 2]
```diff
@@ -68,7 +68,8 @@ impl PyLogger {
     }
 
     fn init(verbosity: Level, py: Python) -> Result<(), SetLoggerError> {
-        let logger = PyLogger::new(py).unwrap();
+        let logger =
+            PyLogger::new(py).expect("Failed to instantiate Python logger; check library paths.");
```
Comment (vaporos) [inline validator/src/pylogger.rs]: Avoid the period here please.

File: validator/src/pylogger.rs (hunk 1 above)
Comment (peterschwarz) [reply validator/src/pylogger.rs]: Yes, it only returns the `SetLoggerError` which is only returned if a logger has been set twice.

Comment (peterschwarz) [reply validator/src/pylogger.rs]: See: https://docs.rs/log/0.4.6/log/struct.SetLoggerError.html

**Pull request merged**